fsspec/implementations/tests/test_reference.py

```diff
@@ -210,3 +212,24 @@ def test_empty():
     pytest.importorskip("jinja2")
     fs = fsspec.filesystem("reference", fo={"version": 1}, target_protocol="http")
     assert fs.references == {}
+
+
+def test_get_sync(tmpdir):
+    localfs = LocalFileSystem()
+
+    real = (tmpdir / "file")
+    real.write_binary(b"0123456789")
+
+    refs = {"a": b"data", "b": (str(real), 0, 5), "c/d": (str(real), 1, 6)}
+    fs = fsspec.filesystem("reference", fo=refs, fs=localfs)
+
+    fs.get("a", str(tmpdir / "a"))
+    assert (tmpdir / "a").read_binary() == b"data"
+    fs.get("b", str(tmpdir / "b"))
+    assert (tmpdir / "b").read_binary() == b"01234"
+    fs.get("c/d", str(tmpdir / "d"))
+    assert (tmpdir / "d").read_binary() == b"123456"
+    fs.get("c", str(tmpdir / "c"))
+    assert (tmpdir / "c").isdir()
+    assert (tmpdir / "c" / "d").read_binary() == b"123456"
```

Comment:
@martindurant Unfortunately I don't have enough context about reffs, looks like there is a bug in it though. If you have any quick tips, I would really appreciate them.
```
>       fs.get("c", str(tmpdir / "c"))

fsspec/implementations/tests/test_reference.py:232:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
fsspec/implementations/reference.py:244: in get
    return AbstractFileSystem.get(self, rpath, lpath, recursive=recursive, **kwargs)
fsspec/spec.py:797: in get
    self.get_file(rpath, lpath, **kwargs)
fsspec/implementations/reference.py:235: in get_file
    data = self.cat_file(rpath, **kwargs)
fsspec/implementations/reference.py:220: in cat_file
    part_or_url, start0, end0 = self._cat_common(path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <fsspec.implementations.reference.ReferenceFileSystem object at 0x1050b9cd0>, path = 'c'

    def _cat_common(self, path):
        path = self._strip_protocol(path)
        logger.debug(f"cat: {path}")
        # TODO: can extract and cache templating here
        if self.dataframe:
            part = self.df.loc[path]
            if part["data"]:
                part = part["data"]
            else:
                part = part[["url", "offset", "size"]]
        else:
>           part = self.references[path]
E           KeyError: 'c'

fsspec/implementations/reference.py:191: KeyError

```

Reply:
```diff
+++ b/fsspec/implementations/reference.py
@@ -3,6 +3,7 @@ import io
 import itertools
 import logging
 from functools import lru_cache
+import os

 import fsspec.core

@@ -227,11 +228,15 @@ class ReferenceFileSystem(AsyncFileSystem):
         self.references[path] = value

     async def _get_file(self, rpath, lpath, **kwargs):
+        if self.isdir(rpath):
+            return os.makedirs(lpath, exist_ok=True)
         data = await self._cat_file(rpath)
         with open(lpath, "wb") as f:
             f.write(data)

     def get_file(self, rpath, lpath, callback=_DEFAULT_CALLBACK, **kwargs):
+        if self.isdir(rpath):
+            return os.makedirs(lpath, exist_ok=True)
         data = self.cat_file(rpath, **kwargs)
         callback.lazy_call("set_size", len, data)```

Reply:
Thank you! 🙏